src/main/java/net/dv8tion/jda/internal/handle/MessageReactionHandler.java

```diff
@@ -126,11 +116,12 @@ protected Long handleInternally(DataObject content)
             channel = api.getPrivateChannelById(channelId);
         if (channel == null)
         {
-            api.getEventCache().cache(EventCache.Type.CHANNEL, channelId, responseNumber, allContent, this::handle);
-            EventCache.LOG.debug("Received a reaction for a channel that JDA does not currently have cached");
-            return null;
+            //create a new private channel with minimal information for this event
+            channel = getJDA().getEntityBuilder().createPrivateChannel(
+                    DataObject.empty()
+                            .put("id", channelId)
+            );
```

Comment:
We definitely cannot just assume this here.
This could be from a channel type we don't support yet, or it could be from a channel we just haven't received yet.
Not sure that I understand why it'd be implemented this way 😕

Reply:
We should at a minimum be able to see if `guild_id` is populated to make a decision about whether to try and create a private channel or use the event cache.